### PR TITLE
enhancement(dev): Convert to indoc blocks

### PR DESCRIPTION
**Pull request by lucperkins**
This is "just" an aesthetic thing but I do think that it helps quite a bit with readability in these files. Hat tip to @pablosichert for the suggestion. This may be overkill, but I figured why not? 🤷🏼‍♂️

### CODE REVIEW

File: benches/remap.rs [hunk 1]
```diff
@@ -68,10 +69,12 @@ fn benchmark_remap(c: &mut Criterion) {
...
             Remap::new(RemapConfig {
-                source: r#".foo = "bar"
-            .bar = "baz"
-            .copy = .copy_from"#
-                    .to_string(),
+                source: indoc! {r#"
+                    .foo = "bar"
+                    .bar = "baz"
+                    .copy = .copy_from
+                    "#}
```
Comment (pablosichert) [inline benches/remap.rs:76]:
```suggestion
                source: indoc! {r#"
                    .foo = "bar"
                    .bar = "baz"
                    .copy = .copy_from
                "#}
```

File: benches/remap.rs [hunk 2]
```diff
@@ -230,15 +233,14 @@ fn benchmark_remap(c: &mut Criterion) {
...
-                   "#,
-                )
+                toml::from_str::<CoercerConfig>(indoc! { r#"
+                        drop_unspecified = false
+
+                        [types]
+                        number = "int"
+                        bool = "bool"
+                        timestamp = "timestamp|%d/%m/%Y:%H:%M:%S %z"
+                   "#})
```
Comment (pablosichert) [inline benches/remap.rs:243]:
```suggestion
                toml::from_str::<CoercerConfig>(indoc! { r#"
                    drop_unspecified = false

                    [types]
                    number = "int"
                    bool = "bool"
                    timestamp = "timestamp|%d/%m/%Y:%H:%M:%S %z"
                "#})
```

File: benches/wasm/mod.rs [hunk 3]
```diff
@@ -28,12 +29,10 @@ pub fn protobuf(c: &mut Criterion) {
...
-                toml::from_str(
-                    r#"
-                module = "tests/data/wasm/protobuf/target/wasm32-wasi/release/protobuf.wasm"
-                artifact_cache = "target/artifacts/"
-                "#,
-                )
+                toml::from_str(indoc! { r#"
+                        module = "tests/data/wasm/protobuf/target/wasm32-wasi/release/protobuf.wasm"
+                        artifact_cache = "target/artifacts/"
+                    "#})
```
Comment (pablosichert) [inline benches/wasm/mod.rs:35]:
```suggestion
                toml::from_str(indoc! { r#"
                    module = "tests/data/wasm/protobuf/target/wasm32-wasi/release/protobuf.wasm"
                    artifact_cache = "target/artifacts/"
                "#})
```

**Comment by pablosichert** [comment]
There's a bit of inconsistency going on the the YAMLs:

```yaml
[foo]
bar = true
```

vs

```yaml
[foo]
  bar = true
```

Might fix that as well while we're at it 😄

**Comment by lucperkins** [comment]
@pablosichert This is why I'm philosophically opposed to indenting TOML 😄 (unlike some others on the team)

File: src/generate.rs [hunk 4]
```diff
@@ -431,166 +431,171 @@ mod tests {
     fn generate_basic() {
         assert_eq!(
             generate_example(true, "stdin/json_parser/console", &None),
-            Ok(r#"data_dir = "/var/lib/vector/"
```
Comment (jszwedko) [inline src/generate.rs:434]: I think you need to import `indoc` in this file (see errors).

**Review by jszwedko — APPROVED** [review]
Nice, I'm a fan.

I don't have strong opinions about indenting vs. not indenting TOML. It would be nice to be consistent, but I don't think that needs to block this.

**Comment by pablosichert** [comment]
Jep, also don't care for which alternative we choose as long as we're consistent with it.

File: src/sinks/humio/metrics.rs [hunk 5]
```diff
@@ -121,39 +123,33 @@ mod tests {
...
-            endpoint = "https://localhost:9200/"
-            encoding = "json"
-            "#,
-        )
+        let (config, _) = load_sink::<HumioMetricsConfig>(indoc! {r#"
+                token = "atoken"
+                batch.max_events = 1
+                endpoint = "https://localhost:9200/"
+                encoding = "json"
+            "#})
```
Comment (pablosichert) [inline src/sinks/humio/metrics.rs:131]:
```suggestion
        let (config, _) = load_sink::<HumioMetricsConfig>(indoc! {r#"
            token = "atoken"
            batch.max_events = 1
            endpoint = "https://localhost:9200/"
            encoding = "json"
        "#})
```

File: src/sinks/humio/metrics.rs [hunk 6]
```diff
@@ -121,39 +123,33 @@ mod tests {
...
-            host = "https://localhost:9200/"
-            encoding = "json"
-            "#,
-        )
+        let (config, _) = load_sink::<HumioMetricsConfig>(indoc! {r#"
+                token = "atoken"
+                batch.max_events = 1
+                host = "https://localhost:9200/"
+                encoding = "json"
+            "#})
```
Comment (pablosichert) [inline src/sinks/humio/metrics.rs:140]:
```suggestion
        let (config, _) = load_sink::<HumioMetricsConfig>(indoc! {r#"
            token = "atoken"
            batch.max_events = 1
            host = "https://localhost:9200/"
            encoding = "json"
        "#})
```

File: src/sinks/humio/metrics.rs [hunk 7]
```diff
@@ -121,39 +123,33 @@ mod tests {
...
-            token = "atoken"
-            batch.max_events = 1
-            encoding = "json"
-            "#,
-        )
+        let (mut config, cx) = load_sink::<HumioMetricsConfig>(indoc! {r#"
+                token = "atoken"
+                batch.max_events = 1
+                encoding = "json"
+            "#})
```
Comment (pablosichert) [inline src/sinks/humio/metrics.rs:152]:
```suggestion
        let (mut config, cx) = load_sink::<HumioMetricsConfig>(indoc! {r#"
            token = "atoken"
            batch.max_events = 1
            encoding = "json"
        "#})
```

File: src/sinks/influxdb/logs.rs [hunk 8]
```diff
@@ -77,14 +78,14 @@ inventory::submit! {
...
-            token = "${INFLUXDB_TOKEN}""#,
-        )
+        toml::from_str(indoc! {r#"
+                endpoint = "http://localhost:8086/"
+                namespace = "my-namespace"
+                tags = []
+                org = "my-org"
+                bucket = "my-bucket"
+                token = "${INFLUXDB_TOKEN}"
+            "#})
```
Comment (pablosichert) [inline src/sinks/influxdb/logs.rs:88]:
```suggestion
        toml::from_str(indoc! {r#"
            endpoint = "http://localhost:8086/"
            namespace = "my-namespace"
            tags = []
            org = "my-org"
            bucket = "my-bucket"
            token = "${INFLUXDB_TOKEN}"
        "#})
```

File: src/sinks/influxdb/logs.rs [hunk 9]
```diff
@@ -488,13 +490,11 @@ mod tests {
...
-            namespace = "ns"
-            endpoint = "http://localhost:9999"
-            database = "my-database"
-        "#,
-        )
+        let (mut config, cx) = load_sink::<InfluxDBLogsConfig>(indoc! {r#"
+                namespace = "ns"
+                endpoint = "http://localhost:9999"
+                database = "my-database"
+            "#})
```
Comment (pablosichert) [inline src/sinks/influxdb/logs.rs:497]:
```suggestion
        let (mut config, cx) = load_sink::<InfluxDBLogsConfig>(indoc! {r#"
            namespace = "ns"
            endpoint = "http://localhost:9999"
            database = "my-database"
        "#})
```

File: src/sinks/influxdb/logs.rs [hunk 10]
```diff
@@ -549,15 +549,13 @@ mod tests {
...
-            token = "my-token"
-        "#,
-        )
+        let (mut config, cx) = load_sink::<InfluxDBLogsConfig>(indoc! {r#"
+                namespace = "ns"
+                endpoint = "http://localhost:9999"
+                bucket = "my-bucket"
+                org = "my-org"
+                token = "my-token"
+            "#})
```
Comment (pablosichert) [inline src/sinks/influxdb/logs.rs:558]:
```suggestion
        let (mut config, cx) = load_sink::<InfluxDBLogsConfig>(indoc! {r#"
            namespace = "ns"
            endpoint = "http://localhost:9999"
            bucket = "my-bucket"
            org = "my-org"
            token = "my-token"
        "#})
```

File: src/sinks/sematext/logs.rs [hunk 11]
```diff
@@ -42,10 +43,10 @@ inventory::submit! {
...
 impl GenerateConfig for SematextLogsConfig {
     fn generate_config() -> toml::Value {
-        toml::from_str(
-            r#"region = "us"
-            token = "${SEMATEXT_TOKEN}""#,
-        )
+        toml::from_str(indoc! {r#"
+                region = "us"
+                token = "${SEMATEXT_TOKEN}"
+            "#})
```
Comment (pablosichert) [inline src/sinks/sematext/logs.rs:49]:
```suggestion
        toml::from_str(indoc! {r#"
            region = "us"
            token = "${SEMATEXT_TOKEN}"
        "#})
```

File: src/sinks/sematext/logs.rs [hunk 12]
```diff
@@ -126,12 +128,10 @@ mod tests {
...
-        let (mut config, cx) = load_sink::<SematextLogsConfig>(
-            r#"
-            region = "us"
-            token = "mylogtoken"
-        "#,
-        )
+        let (mut config, cx) = load_sink::<SematextLogsConfig>(indoc! {r#"
+                region = "us"
+                token = "mylogtoken"
+                "#})
```
Comment (pablosichert) [inline src/sinks/sematext/logs.rs:134]:
```suggestion
        let (mut config, cx) = load_sink::<SematextLogsConfig>(indoc! {r#"
            region = "us"
            token = "mylogtoken"
        "#})
```

File: src/sinks/sematext/metrics.rs [hunk 13]
```diff
@@ -45,11 +46,11 @@ inventory::submit! {
...
-        toml::from_str(
-            r#"region = "us"
-            default_namespace = "vector"
-            token = "${SEMATEXT_TOKEN}""#,
-        )
+        toml::from_str(indoc! {r#"
+                region = "us"
+                default_namespace = "vector"
+                token = "${SEMATEXT_TOKEN}"
+            "#})
```
Comment (pablosichert) [inline src/sinks/sematext/metrics.rs:53]:
```suggestion
        toml::from_str(indoc! {r#"
            region = "us"
            default_namespace = "vector"
            token = "${SEMATEXT_TOKEN}"
        "#})
```

File: src/sinks/sematext/metrics.rs [hunk 14]
```diff
@@ -326,14 +328,12 @@ mod tests {
...
-            token = "atoken"
-            batch.max_events = 1
-            "#,
-        )
+        let (mut config, cx) = load_sink::<SematextMetricsConfig>(indoc! {r#"
+                region = "eu"
+                default_namespace = "ns"
+                token = "atoken"
+                batch.max_events = 1
+            "#})
```
Comment (pablosichert) [inline src/sinks/sematext/metrics.rs:336]:
```suggestion
        let (mut config, cx) = load_sink::<SematextMetricsConfig>(indoc! {r#"
            region = "eu"
            default_namespace = "ns"
            token = "atoken"
            batch.max_events = 1
        "#})
```

File: src/sinks/humio/metrics.rs [hunk 15]
```diff
@@ -53,11 +54,11 @@ inventory::submit! {
...
-        toml::from_str(
-            r#"host_key = "hostname"
-            token = "${HUMIO_TOKEN}"
-            encoding.codec = "json""#,
-        )
+        toml::from_str(indoc! {r#"
+                host_key = "hostname"
+                token = "${HUMIO_TOKEN}"
+                encoding.codec = "json"
+            "#})
```
Comment (pablosichert) [inline src/sinks/humio/metrics.rs:61]:
```suggestion
        toml::from_str(indoc! {r#"
            host_key = "hostname"
            token = "${HUMIO_TOKEN}"
            encoding.codec = "json"
        "#})
```

File: src/rusoto/region.rs [hunk 16]
```diff
@@ -110,12 +111,10 @@ mod tests {
...
-        let config: Config = toml::from_str(
-            r#"
-        [inner]
-        region = "us-east-1"
-        "#,
-        )
+        let config: Config = toml::from_str(indoc! {r#"
+                [inner]
+                region = "us-east-1"
+            "#})
```
Comment (pablosichert) [inline src/rusoto/region.rs:117]:
```suggestion
        let config: Config = toml::from_str(indoc! {r#"
            [inner]
            region = "us-east-1"
        "#})
```

File: src/rusoto/region.rs [hunk 17]
```diff
@@ -124,12 +123,10 @@ mod tests {
...
-        let config: Config = toml::from_str(
-            r#"
-        [inner]
-        endpoint = "http://localhost:9000"
-        "#,
-        )
+        let config: Config = toml::from_str(indoc! {r#"
+                [inner]
+                endpoint = "http://localhost:9000"
+            "#})
```
Comment (pablosichert) [inline src/rusoto/region.rs:129]:
```suggestion
        let config: Config = toml::from_str(indoc! {r#"
            [inner]
            endpoint = "http://localhost:9000"
        "#})
```

File: src/rusoto/region.rs [hunk 18]
```diff
@@ -143,12 +140,10 @@ mod tests {
...
-        let config: Config = toml::from_str(
-            r#"
-        [inner]
-        endpoint_is_spelled_wrong = "http://localhost:9000"
-        "#,
-        )
+        let config: Config = toml::from_str(indoc! {r#"
+                [inner]
+                endpoint_is_spelled_wrong = "http://localhost:9000"
+            "#})
```
Comment (pablosichert) [inline src/rusoto/region.rs:146]:
```suggestion
        let config: Config = toml::from_str(indoc! {r#"
            [inner]
            endpoint_is_spelled_wrong = "http://localhost:9000"
        "#})
```

File: src/sinks/datadog/logs.rs [hunk 19]
```diff
@@ -59,10 +60,10 @@ inventory::submit! {
...
 impl GenerateConfig for DatadogLogsConfig {
     fn generate_config() -> toml::Value {
-        toml::from_str(
-            r#"api_key = "${DATADOG_API_KEY_ENV_VAR}"
-            encoding.codec = "json""#,
-        )
+        toml::from_str(indoc! {r#"
+                api_key = "${DATADOG_API_KEY_ENV_VAR}"
+                encoding.codec = "json"
+            "#})
```
Comment (pablosichert) [inline src/sinks/datadog/logs.rs:66]:
```suggestion
        toml::from_str(indoc! {r#"
            api_key = "${DATADOG_API_KEY_ENV_VAR}"
            encoding.codec = "json"
        "#})
```

File: src/sinks/datadog/logs.rs [hunk 20]
```diff
@@ -305,14 +307,12 @@ mod tests {
...
-            compression = "none"
-            batch.max_events = 1
-            "#,
-        )
+        let (mut config, cx) = load_sink::<DatadogLogsConfig>(indoc! {r#"
+                api_key = "atoken"
+                encoding = "text"
+                compression = "none"
+                batch.max_events = 1
+            "#})
```
Comment (pablosichert) [inline src/sinks/datadog/logs.rs:315]:
```suggestion
        let (mut config, cx) = load_sink::<DatadogLogsConfig>(indoc! {r#"
            api_key = "atoken"
            encoding = "text"
            compression = "none"
            batch.max_events = 1
        "#})
```

File: src/sinks/datadog/logs.rs [hunk 21]
```diff
@@ -340,14 +340,12 @@ mod tests {
...
-            compression = "none"
-            batch.max_events = 1
-            "#,
-        )
+        let (mut config, cx) = load_sink::<DatadogLogsConfig>(indoc! {r#"
+                api_key = "atoken"
+                encoding = "json"
+                compression = "none"
+                batch.max_events = 1
+            "#})
```
Comment (pablosichert) [inline src/sinks/datadog/logs.rs:348]:
```suggestion
        let (mut config, cx) = load_sink::<DatadogLogsConfig>(indoc! {r#"
            api_key = "atoken"
            encoding = "json"
            compression = "none"
            batch.max_events = 1
        "#})
```

File: src/sinks/gcp/cloud_storage.rs [hunk 22]
```diff
@@ -146,11 +147,11 @@ inventory::submit! {
...
-        toml::from_str(
-            r#"bucket = "my-bucket"
-            credentials_path = "/path/to/credentials.json"
-            encoding.codec = "ndjson""#,
-        )
+        toml::from_str(indoc! {r#"
+                bucket = "my-bucket"
+                credentials_path = "/path/to/credentials.json"
+                encoding.codec = "ndjson"
+            "#})
```
Comment (pablosichert) [inline src/sinks/gcp/cloud_storage.rs:154]:
```suggestion
        toml::from_str(indoc! {r#"
            bucket = "my-bucket"
            credentials_path = "/path/to/credentials.json"
            encoding.codec = "ndjson"
        "#})
```

File: src/sinks/gcp/pubsub.rs [hunk 23]
```diff
@@ -211,12 +212,10 @@ mod tests {
...
-        let config: PubsubConfig = toml::from_str(
-            r#"
-           project = "project"
-           topic = "topic"
-        "#,
-        )
+        let config: PubsubConfig = toml::from_str(indoc! {r#"
+                project = "project"
+                topic = "topic"
+            "#})
```
Comment (pablosichert) [inline src/sinks/gcp/pubsub.rs:218]:
```suggestion
        let config: PubsubConfig = toml::from_str(indoc! {r#"
            project = "project"
            topic = "topic"
        "#})
```

File: src/sinks/gcp/stackdriver_logs.rs [hunk 24]
```diff
@@ -282,15 +283,13 @@ mod tests {
...
-           encoding.except_fields = ["anumber"]
-        "#,
-        )
+        let config: StackdriverConfig = toml::from_str(indoc! {r#"
+                project_id = "project"
+                log_id = "testlogs"
+                resource.type = "generic_node"
+                resource.namespace = "office"
+                encoding.except_fields = ["anumber"]
+            "#})
```
Comment (pablosichert) [inline src/sinks/gcp/stackdriver_logs.rs:292]:
```suggestion
        let config: StackdriverConfig = toml::from_str(indoc! {r#"
            project_id = "project"
            log_id = "testlogs"
            resource.type = "generic_node"
            resource.namespace = "office"
            encoding.except_fields = ["anumber"]
        "#})
```

File: src/sinks/gcp/stackdriver_logs.rs [hunk 25]
```diff
@@ -314,14 +313,12 @@ mod tests {
...
-           resource.type = "generic_node"
-           resource.namespace = "office"
-        "#,
-        )
+        let config: StackdriverConfig = toml::from_str(indoc! {r#"
+                project_id = "project"
+                log_id = "testlogs"
+                resource.type = "generic_node"
+                resource.namespace = "office"
+            "#})
```
Comment (pablosichert) [inline src/sinks/gcp/stackdriver_logs.rs:321]:
```suggestion
        let config: StackdriverConfig = toml::from_str(indoc! {r#"
            project_id = "project"
            log_id = "testlogs"
            resource.type = "generic_node"
            resource.namespace = "office"
        "#})
```

File: src/sinks/gcp/stackdriver_logs.rs [hunk 26]
```diff
@@ -376,14 +373,12 @@ mod tests {
...
```
Comment (pablosichert) [inline src/sinks/gcp/stackdriver_logs.rs:381]:
```suggestion
        let config: StackdriverConfig = toml::from_str(indoc! {r#"
            project_id = "project"
            log_id = "testlogs"
            resource.type = "generic_node"
            resource.namespace = "office"
        "#})
```

File: src/sinks/gcp/stackdriver_logs.rs [hunk 27]
```diff
@@ -444,14 +439,12 @@ mod tests {
...
```
Comment (pablosichert) [inline src/sinks/gcp/stackdriver_logs.rs:447]:
```suggestion
        let config: StackdriverConfig = toml::from_str(indoc! {r#"
            project_id = "project"
            log_id = "testlogs"
            resource.type = "generic_node"
            resource.namespace = "office"
        "#})
```

File: src/sinks/gcp/stackdriver_logs.rs [hunk 28]
```diff
@@ -460,24 +453,20 @@ mod tests {
...
-           log_id = "testlogs"
-           resource.type = "generic_node"
-           resource.namespace = "office"
-        "#,
-        )
+        toml::from_str::<StackdriverConfig>(indoc! {r#"
+                log_id = "testlogs"
+                resource.type = "generic_node"
+                resource.namespace = "office"
+            "#})
```
Comment (pablosichert) [inline src/sinks/gcp/stackdriver_logs.rs:460]:
```suggestion
        toml::from_str::<StackdriverConfig>(indoc! {r#"
            log_id = "testlogs"
            resource.type = "generic_node"
            resource.namespace = "office"
        "#})
```

File: src/sinks/gcp/stackdriver_logs.rs [hunk 29]
```diff
@@ -460,24 +453,20 @@ mod tests {
...
-           resource.namespace = "office"
-        "#,
-        )
+        toml::from_str::<StackdriverConfig>(indoc! {r#"
+                project_id = "project"
+                folder_id = "folder"
+                log_id = "testlogs"
+                resource.type = "generic_node"
+                resource.namespace = "office"
+            "#})
```
Comment (pablosichert) [inline src/sinks/gcp/stackdriver_logs.rs:469]:
```suggestion
        toml::from_str::<StackdriverConfig>(indoc! {r#"
            project_id = "project"
            folder_id = "folder"
            log_id = "testlogs"
            resource.type = "generic_node"
            resource.namespace = "office"
        "#})
```

**Comment by pablosichert** [comment]
My backspace key is on fire. Sorry for the spam, thought it was just a couple of occurrences and then it was already too late to not keep going 😄

**Comment by lukesteensen** [comment]
> I'm philosophically opposed to indenting TOML smile (unlike some others on the team)

I'm also against indenting TOML, just in case it helps break any ties :smile:

**Comment by lucperkins** [comment]
@pablosichert I may need to do a little test here because I feel like `cargo fmt` may be overwriting some of my changes. I'll get back to you!

@lukesteensen We are legion! I will keep your vote in mind for a future conflagration.

File: lib/prometheus-parser/src/lib.rs [hunk 30]
```diff
@@ -236,10 +236,11 @@ pub fn group_metrics(input: &str) -> Result<Vec<MetricGroup>, ParserError> {
 #[cfg(test)]
 mod test {
     use super::*;
+    use indoc::indoc;
 
     #[test]
     fn test_group_metrics() {
-        let input = r##"
+        let input = indoc! {r#"
```
Comment (bruceg) [inline lib/prometheus-parser/src/lib.rs:243]: Other than uniformity, I don't see what this usage benefits this code.

File: lib/prometheus-parser/src/line.rs [hunk 31]
```diff
@@ -619,7 +620,7 @@ mod test {
 
     #[test]
     fn test_parse_line() {
-        let input = r##"
+        let input = indoc! {r#"
```
Comment (bruceg) [inline lib/prometheus-parser/src/line.rs:623]: Ditto here

File: src/sinks/influxdb/logs.rs [hunk 32]
```diff
@@ -259,13 +261,13 @@ mod tests {
 
     #[test]
     fn test_config_without_tags() {
-        let config = r#"
+        let config = indoc! {r#"
```
Comment (bruceg) [inline src/sinks/influxdb/logs.rs:264]: This too doesn't seem to benefit much from `indoc`.

File: src/sinks/util/encoding/mod.rs [hunk 33]
```diff
@@ -174,21 +175,21 @@ mod tests {
         PathIter::new(a).collect()
     }
 
-    const TOML_SIMPLE_STRING: &str = r#"
+    const TOML_SIMPLE_STRING: &str = indoc! {r#"
```
Comment (bruceg) [inline src/sinks/util/encoding/mod.rs:178]: Same, seems like little benefit here and below in this file.

**Review by bruceg — APPROVED** [review]
This looks like a win overall. There are a few conversions that I commented on that don't seem to be adding anything (neither fixing indentation nor making the start/end quotes less awkward). That's not blocking though.

File: src/sinks/sematext/metrics.rs (hunk 13 above)
Comment (pablosichert) [reply src/sinks/sematext/metrics.rs:56]: So did `fmt` change this back again? 🤔

**Pull request merged**